locksmithctl/daemon.go

```
@@ -116,6 +117,54 @@ func expBackoff(interval time.Duration) time.Duration {
 	return interval
 }
 
+// getPeriodicAndSleep get the REBOOT_WINDOW_START, REBOOT_WINDOW_LENGTH
+// and check if everything is ok and the get then Periodic. Check if
+// it is outside of the reboot window and If yes then sleep.
+func getPeriodicAndSleep() (err error) {
+	var period *timeutil.Periodic
+
+	startw := os.Getenv("LOCKSMITHD_REBOOT_WINDOW_START")
+	if startw == "" {
+		startw = os.Getenv("REBOOT_WINDOW_START")
+	}
+
+	lengthw := os.Getenv("LOCKSMITHD_REBOOT_WINDOW_LENGTH")
+	if lengthw == "" {
+		lengthw = os.Getenv("REBOOT_WINDOW_LENGTH")
+	}
+
+	if (startw == "") != (lengthw == "") {
+		return errors.New("Either both or neither $REBOOT_WINDOW_START and $REBOOT_WINDOW_LENGTH must be set")
+	}
+
+	if startw != "" && lengthw != "" {
+		p, err := timeutil.ParsePeriodic(startw, lengthw)
+		if err != nil {
+			dlog.Fatalf("Error parsing reboot window: %s", err)
```

same as above  - we can now return an error instead of log it. :)